tests/std/tests/P0355R7_calendars_and_time_zones_formatting/test.cpp

```diff
@@ -332,6 +332,32 @@ void test_month_formatter() {
     stream_helper(STR("20 is not a valid month"), month{20});
 }
 
+template <typename CharT>
+void test_year_formatter() {
+    assert(format(STR("{}"), year{0}) == STR("0000"));
+    assert(format(STR("{}"), year{-200}) == STR("-0200"));
+    assert(format(STR("{}"), year{121}) == STR("0121"));
+
+    assert(format(STR("{:%Y %y%C}"), year{1912}) == STR("1912 1219"));
+    assert(format(STR("{:%Y %y%C}"), year{-1912}) == STR("-1912 88-20"));
+    // TRANSITION, add tests for EY Oy Ey EC
+
+    stream_helper(STR("1900"), year{1900});
+    stream_helper(STR("2000"), year{2000});
+    stream_helper(STR("-32768 is not a valid year"), year{-32768});
+}
+
+template <typename CharT>
+void test_year_month_day_formatter() {
+    auto invalid = year_month_day{year{1900}, month{0}, day{1}};

```

Comment:
We conventionally avoid using `auto` with a temporary like this - it's simpler to directly name the type:
```suggestion
    year_month_day invalid{year{1900}, month{0}, day{1}};
```